packages/shopping-cart/src/with-shopping-cart.tsx

```diff
@@ -1,12 +1,23 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import ShoppingCartContext from './shopping-cart-context';
 import useShoppingCart from './use-shopping-cart';
 import type { WithShoppingCartProps } from './types';
 
 export default function withShoppingCart< ComponentProps >(
-	Component: React.ComponentType< ComponentProps & WithShoppingCartProps >
+	Component: React.ComponentType< ComponentProps & WithShoppingCartProps >,
+	mapPropsToCartKey?: ( props: ComponentProps ) => string | undefined
```

Comment:
Agreed, but if we change the API of `withShoppingCart` that drastically we'd need to also change every instance of it in this PR, and my goal was to put that off until a later PR since it's a lot of additional code changes. Changing the API like this could be done also in a later PR.

Reply:
OK, I'm going to take a deeper look at the `useCartKey` and `useShoppingCart` hooks and how they work together, either through the `defaultCartKey` or through the mapping function.